aptos-move/framework/aptos-framework/sources/vesting_without_staking.move

```diff
@@ -0,0 +1,1132 @@
+        let last_completed_period =
+            (timestamp::now_seconds() - vesting_schedule.start_timestamp_secs) / vesting_schedule.period_duration;
+        if (last_completed_period < next_period_to_vest) {
+            return
+        };
+
+        // Index is 0-based while period is 1-based so we need to subtract 1.
+        let schedule = &vesting_schedule.schedule;
+        let schedule_index = next_period_to_vest - 1;
+        let vesting_fraction = if (schedule_index < vector::length(schedule)) {
+            *vector::borrow(schedule, schedule_index)
+        } else {
+            // Last vesting schedule fraction will repeat until the grant runs out.
+            *vector::borrow(schedule, vector::length(schedule) - 1)
+        };
+
+        vesting_schedule.last_vested_period = next_period_to_vest;
+
+        emit_event(
+            &mut vesting_contract.vest_events,
+            VestEvent {
+                admin: vesting_contract.admin,
+                vesting_contract_address: contract_address,
+                period_vested: next_period_to_vest,
+            },
+        );
+        // Every shareholder should receive the money in proportion to their shares.
+        distribute(vesting_fraction, contract_address);
+    }
+
+    /// Distribute any withdrawable grant.
+    /// This is no entry function anymore as it's called from within the vest.
+    fun distribute(vesting_fraction: FixedPoint32, contract_address: address) acquires VestingContract {
+        assert_active_vesting_contract(contract_address);
+
+        let vesting_contract = borrow_global_mut<VestingContract>(contract_address);
+        let vesting_signer = get_vesting_account_signer_internal(vesting_contract);
+        let shareholders_address = simple_map::keys(&vesting_contract.shareholders);
+        let total_amount_left = 0;
+        // Distribute coins to shareholders.
+        vector::for_each_ref(&shareholders_address, |shareholder| {
+            let shareholder = *shareholder;
+            let amount = min(simple_map::borrow(& vesting_contract.shareholders, &shareholder).left_amount,fixed_point32::multiply_u64(simple_map::borrow(& vesting_contract.shareholders, &shareholder).init_amount, vesting_fraction));
+            let recipient_address = get_beneficiary(vesting_contract, shareholder);
+            coin::transfer<AptosCoin>(&vesting_signer, recipient_address, amount);
+            let shareholder_amount = simple_map::borrow_mut(&mut vesting_contract.shareholders, &shareholder);
+            shareholder_amount.left_amount = shareholder_amount.left_amount - amount;
+            total_amount_left = total_amount_left + shareholder_amount.left_amount;
+        });
+        let total_balance = coin::balance<AptosCoin>(contract_address);
+        assert!(total_amount_left == total_balance, EBALANCE_THE_SAME);
+        if (total_balance == 0) {
+            let coins = coin::withdraw<AptosCoin>(&vesting_signer, 0);
+            coin::destroy_zero(coins);
```

Comment:
Ok, I think this was my mistake. Doing a `withdraw` creates a new instance of a coin with the `amount`, and then we call `destroy_zero`, which is poitless. These two lines can be removed.

Reply:
Removed!